components/emoji_picker/components/emoji_picker_kaomoji.tsx

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ChangeEvent, KeyboardEvent, useEffect, useRef, useState} from 'react';
```

Comment:
Could you pull this out and submit it as a separate PR? I'm 0/5 on if we add it, but this PR's already massive, so we should try to pare it down wherever possible